### DISCUSSION TITLE
XMPP (RFC 6120), PubSub (XEP-0060) and Atom Over XMPP (RFC 4287)

### DISCUSSION THREAD

**Discussion by sjehuda** [Athou/commafeed#1364] [Ideas]
Greetings!

My name is Schimon and I am a champion of Syndication and XMPP.

For those of you who are not famiiliar, XMPP (previously Jabber) is a messaging system with open standards. XMPP is extensible via XEP.

I will discuss about XEP-0060: Publish-Subscribe.

Almost a couple of decades ago, a [discussion](https://datatracker.ietf.org/meeting/66/materials/slides-66-atompub-1.pdf) arose about incorporating Atom (RFC 4287) with XMPP PubSub, which is really easy because both are XML data.

One of the major benefits of utilizing PubSub is native Push Notifications (yes, that technique was much before HTML5) which do not require HTTP pulling.

That would be a great feature to have CommaFeed to utilize XMPP PubSub natively!

See a [proof of concept](https://codeberg.org/poezio/slixmpp/issues/3526) I have made to parse PubSub nodes as Atom Syndication Format.

https://datatracker.ietf.org/meeting/66/materials/slides-66-atompub-1.pdf
https://codeberg.org/poezio/slixmpp/issues/3526
https://github.com/igniterealtime/Smack (XMPP for Java)

Cheers,
Schimon

**Comment by Athou** [comment]
Hi @sjehuda,

I'm familiar with XMPP but I've never seen it used to push Atom content. Are there any sites using this to publish their content?

[WebSub](https://www.w3.org/TR/websub/) offers a similar solution as a PubSub model and is used by some sites, although CommaFeed does not implement it (yet).

**Reply by sjehuda** [reply]
> Are there any sites using this to publish their content?

Yes, Libervia and Movim.

> WebSub offers a similar solution as a PubSub model and is used by some sites, although CommaFeed does not implement it (yet).

I have wrote arguments at https://github.com/fossar/selfoss/issues/1483#issuecomment-2078709443 (PHP)

Thankfully, @igniterealtime and @tigase have made XMPP Java modules with comprehensive support to various of XEPs, including PubSub.

You can use the [script](https://codeberg.org/poezio/slixmpp/issues/3526) I have wrote to convert XMPP PubSub nodes to XML (Atom Syndication Format).

**Reply by sjehuda** [reply]
The proof of concept script has been developed into an HTTP gateway.
I encourage you to view the video demonstration (length 7:44).

https://video.xmpp-it.net/w/vNqcMooy3pqRAZ8Yb8grr1
https://git.xmpp-it.net/sch/Rivista
